saleor/graphql/channel/dataloaders.py

```diff
@@ -24,25 +24,26 @@ def batch_load(self, keys):
         return [channels.get(slug) for slug in keys]
 
 
-class ChannelByOrderLineIdLoader(DataLoader):
-    context_key = "channel_by_orderline"
+class ChannelByOrderIdLoader(DataLoader):
+    context_key = "channel_by_order"
 
     def batch_load(self, keys):
-        def channel_by_lines(order_lines):
-            order_ids = [line.order_id for line in order_lines]
-
-            def channels_by_checkout(orders):
-                channel_ids = [order.channel_id for order in orders]
-
-                return ChannelByIdLoader(self.context).load_many(channel_ids)
-
+        def with_orders(orders):
+            def with_channels(channels):
+                channel_map = {channel.id: channel for channel in channels}
+                return [
+                    channel_map.get(order.channel_id) if order else None
+                    for order in orders
+                ]
+
+            channel_ids = set(order.channel_id if order else None for order in orders)
```

Comment:
Do we need to pass `None` here? 
Why not:
```suggestion
            channel_ids = set(order.channel_id for order in orders if order)
```

Reply:
You are right